app/uk/gov/hmrc/fileupload/controllers/EnvelopeChecker.scala

```diff
@@ -20,46 +20,90 @@ import cats.data.Xor
+import uk.gov.hmrc.fileupload.transfer.TransferService._
 import uk.gov.hmrc.fileupload.utils.StreamsConverter
 
 import scala.concurrent.{ExecutionContext, Future}
 
 object EnvelopeChecker {
 
-  type WithValidEnvelope = EnvelopeId => EssentialAction => EssentialAction
+  type FileSize = Int
+  type WithValidEnvelope = EnvelopeId => (FileSize => EssentialAction) => EssentialAction
 
   import uk.gov.hmrc.fileupload.utils.StreamImplicits.materializer
 
-  def withValidEnvelope(check: (EnvelopeId) => Future[EnvelopeStatusResult])
+  val defaultFileSize = 10 * 1024 * 1024
+
+  def withValidEnvelope(checkEnvelopeDetails: (EnvelopeId) => Future[EnvelopeDetailResult])
                        (envelopeId: EnvelopeId)
-                       (action: EssentialAction)
+                       (action: FileSize => EssentialAction)
                        (implicit ec: ExecutionContext) =
     EssentialAction { implicit rh =>
       Accumulator.flatten {
-        check(envelopeId).map {
-          case Xor.Right("OPEN") =>
-            action(rh)
-          case Xor.Right(otherStatus) =>
-            logAndReturn(LOCKED, s"Unable to upload to envelope: $envelopeId with status: $otherStatus")
-          case Xor.Left(EnvelopeStatusNotFoundError(_)) =>
+        checkEnvelopeDetails(envelopeId).map {
+          case Xor.Right(envelope) =>
+            val status = (envelope \ "status").as[String]
+            if (status == "OPEN") {
+              action(setMaxFileSize(envelope))(rh)
+            } else if (status == "CLOSED") {
+              logAndReturn(LOCKED, s"Unable to upload to envelope: $envelopeId with status: $status")
+            } else if(status == "SEALED") {
+              logAndReturn(LOCKED, s"Unable to upload to envelope: $envelopeId with status: $status")
+            } else {
+              //If uploading to deleted Envelope
+              logAndReturn(BAD_REQUEST, s"Unable to upload to envelope: $envelopeId with status: $status")
+            }
+          case Xor.Left(EnvelopeDetailNotFoundError(_)) =>
             logAndReturn(NOT_FOUND, s"Unable to upload to nonexistent envelope: $envelopeId")
           case Xor.Left(error) =>
             logAndReturn(INTERNAL_SERVER_ERROR, error.toString)
         }
       }
     }
 
+  def setMaxFileSize(envelope: JsValue): FileSize = {
+    val definedConstraints = (envelope \ "constraints").asOpt[Constraints]
+       definedConstraints match {
+           //TODO Possibly in the future this may change to take into account the envelope size e.g. if envelope size is 5MB and the upload limit is 10MB
+         case Some(constraints) => constraints.maxSizePerItem match {
+           case Some(maxSizePerItem) =>
+             val fileSize = maxSizePerItem.replaceAll("[^\\d.]", "").toInt
+             val fileSizeType = maxSizePerItem.toUpperCase.replaceAll("[^KMB]", "")
+             fileSizeType match {
+               case "KB" =>
+                 if(fileSize <= 1024) {
```

Comment:
you might want to use [guard expressions](https://www.safaribooksonline.com/library/view/scala-cookbook/9781449340292/ch03s14.html) here to replace those if else statements to improve readability.